onnxruntime/core/framework/stream_execution_context.h

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#pragma once
+#include "core/common/logging/logging.h"
+#include "core/framework/device_stream_collection.h"
+#include "core/framework/execution_frame.h"
+#include "core/framework/ort_value.h"
+#include "core/framework/iexecutor.h"
+#include "core/framework/stream_handles.h"
+#include "core/graph/basic_types.h"
+#include "core/common/inlined_containers.h"
+#include "core/framework/memory_info.h"
+#ifdef ENABLE_TRAINING
+#include "core/framework/partial_graph_execution_state.h"
+#endif
+
+namespace onnxruntime {
+class SessionState;
+
+class SessionScope;
+typedef InlinedHashMap<std::string, OrtValue> OrtValueCache;
+typedef std::shared_ptr<OrtValueCache> OrtValueCachePtr;
```

Comment:
this part is migrated from the existing code in training build. Let me double check their usage to see whether i can simplify it.

Reply:
I have double checked with training team, first, yes there is no ownership transfer happened so we should be able to avoid shared ptr. second, actually this feature is deprecated, they have a new implementation to support this requirement and there is no custom using it now, so we should be able to simply remove this. 
I would prefer to remove this "cache" here to reduce the complexity. But i prefer to do it in a separate PR, to avoid mixed the scope.